### PR TITLE
8199594: Add doc describing how (?x) ignores spaces in character classes

### PR DESCRIPTION
**Pull request by igraves**
Clarifying note on comments mode to explicitly note that whitespace within character classes is ignored.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Integration blocker
&nbsp;⚠️ The change requires a CSR request to be approved.

### Issue
 * [JDK-8199594](https://bugs.openjdk.java.net/browse/JDK-8199594): Add doc describing how (?x) ignores spaces in character classes

### Reviewers
 * [Brian Burkhalter](https://openjdk.java.net/census#bpb) (@bplb - **Reviewer**) ⚠️ Review applies to 8ab051aa9ecd67fffbb0ebdc842a7d124b3e3193

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3577/head:pull/3577` \
`$ git checkout pull/3577`

Update a local copy of the PR: \
`$ git checkout pull/3577` \
`$ git pull https://git.openjdk.java.net/jdk pull/3577/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3577`

View PR using the GUI difftool: \
`$ git pr show -t 3577`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3577.diff">https://git.openjdk.java.net/jdk/pull/3577.diff</a>

</details>

### CODE REVIEW

**Review by bplb — APPROVED** [review]
Looks all right. Probably a CSR is in order.

**Comment by igraves** [comment]
Will put a /csr in to be safe.

File: src/java.base/share/classes/java/util/regex/Pattern.java [hunk 1]
```diff
@@ -755,6 +755,12 @@
  *    within a group; in the latter case, flags are restored at the end of the
  *    group just as in Perl.  </p></li>
  *
+ *    <li><p> In Perl, <i>free-spacing mode</i> (<i>comments mode</i> in this
+ *    class) denoted by {@code (?x)} will not ignore whitespace inside of
+ *    character classes. In this class, whitespace inside of character classes
+ *    must be escaped to be considered as part of the regular expression when in
+ *    comments mode.  </p></li>
```
Comment (stuart-marks) [inline src/java.base/share/classes/java/util/regex/Pattern.java:762]: Editorial: the run of italicized words makes this a bit hard to follow. Suggest:

    In Perl, <i>free-spacing mode</i> (which is called <i>comments mode</i> in this class)

File: src/java.base/share/classes/java/util/regex/Pattern.java [hunk 2]
```diff
@@ -820,6 +826,10 @@
      *
      * <p> Comments mode can also be enabled via the embedded flag
      * expression&nbsp;{@code (?x)}.
+     *
+     * <p> Note that comments mode ignores whitespace within a character class
+     * contained in a pattern string. Such whitespace needs to be escaped
+     * in order to be treated as if comments mode were not enabled. </p>
```
Comment (stuart-marks) [inline src/java.base/share/classes/java/util/regex/Pattern.java:832]: I think this is good, but 1) it would probably be better placed in the "In this mode" paragraph above, around line 825; and 2) it's normative so it shouldn't say "Note that" (which makes it sound informative).

I'd also reword the second sentence a bit, something like

    Such whitespace needs to be escaped in order to be considered significant.

**Review by stuart-marks — CHANGES_REQUESTED** [review]
A few minor wording adjustments. Please update the CSR accordingly and I'll review it too.

**Comment by igraves** [comment]
Wording updated per @stuart-marks's suggestions. Thanks! CSR updated accordingly.
